tests/test_extractor_tfds.py

```diff
@@ -27,16 +23,50 @@ class TfdsDatasetsTest(TestCase):
     def test_metadata(self):
         env = Environment()
 
-        for metadata in AVAILABLE_TFDS_DATASETS.values():
-            assert metadata.default_converter_name in env.converters
+        for dataset in AVAILABLE_TFDS_DATASETS.values():
+            assert isinstance(dataset.metadata.human_name, str)
+            assert dataset.metadata.human_name != ""
+
+            assert dataset.metadata.default_output_format in env.converters
+
+            assert issubclass(dataset.metadata.media_type, MediaElement)
+
+            # The home URL is optional, but currently every dataset has one.
+            assert isinstance(dataset.metadata.home_url, str)
+            assert dataset.metadata.home_url != ""
+
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_remote_metadata(self):
+        with mock_tfds_data():
+            dataset = AVAILABLE_TFDS_DATASETS["mnist"]
+
+            remote_meta = dataset.query_remote_metadata()
+
+            # verify that the remote metadata contains a copy of the local metadata
+            for attribute in dataset.metadata.__attrs_attrs__:
```

Comment:
Consider using https://www.attrs.org/en/stable/api.html#attrs.fields_dict

Reply:
How would it help me? I don't need to index the attributes, only to iterate over them.

Reply:
I don't think accessing `__attrs_attrs__` is a problem, since it is a [documented](https://www.attrs.org/en/stable/extending.html) attribute.